hudi-common/src/main/java/org/apache/hudi/internal/schema/convert/AvroInternalSchemaConverter.java

```diff
@@ -151,11 +163,11 @@ public static Schema nullableSchema(Schema schema) {
    * @param schema a avro schema.
    * @return a hudi type.
    */
-  public static Type buildTypeFromAvroSchema(Schema schema) {
+  public static Type buildTypeFromAvroSchema(Schema schema, Map<String, Integer> existingNameToPositions) {
     // set flag to check this has not been visited.
-    Deque<String> visited = new LinkedList();
-    AtomicInteger nextId = new AtomicInteger(1);
-    return visitAvroSchemaToBuildType(schema, visited, true, nextId);
+    Deque<String> visited = new LinkedList<>();
+    AtomicInteger nextId = new AtomicInteger(0);
```

Comment:
why do we go from 1->0? Is this because we remove
```
 if (firstVisitRoot) {
          nextAssignId = 0;
        }
 ```

Reply:
I thought this was a bug since you typically start with 0 when coding

Reply:
Also yes, I think it was confusing in general that nextAssignId is set to 0 yet this code is saying it should start at 1